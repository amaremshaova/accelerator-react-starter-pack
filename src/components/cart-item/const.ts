import { ProductsInCart } from '../../types/cart';

export const getTotal = (guitarsInCart : ProductsInCart[]) => {
  let total = 0;
  guitarsInCart.forEach((item) => total += item.product.price * item.count);

  return total;
};
