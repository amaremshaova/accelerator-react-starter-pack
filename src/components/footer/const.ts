export enum SocialName {
  Instagram = 'instagram',
  Facebook = 'facebook',
  Twitter = 'twitter'
}

export enum SocialLink {
  Instagram = 'https://www.instagram.com/',
  Facebook = 'https://www.facebook.com/',
  Twitter = 'https://www.twitter.com/'
}

export enum Contact {
  City = 'Санкт-Петербург',
  Phone = '8-812-500-50-50',
  Metro = 'м. Невский проспект',
  Address = 'ул. Казанская 6',
  OpeningHours = 'с 11:00 до 20:00',
  NumberWorkingDays = 'без выходных'
}
