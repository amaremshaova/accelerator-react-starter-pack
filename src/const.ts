export enum ApiPath{
  Guitars = '/guitars/',
  Comments = '/comments',
  Coupons = '/coupons',
  Orders = '/orders',
}


export enum AppRoute {
  Main = '#',
  Catalog = '/catalog/',
  CatalogPage = '/catalog/page_',
  Empty = '#',
  CatalogStartPage = '/catalog/page_1',
  Cart = '/cart',
  Undefined = '/*',
  Root = '/',
}

