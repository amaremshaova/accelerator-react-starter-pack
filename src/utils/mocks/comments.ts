export const commentsCount = 10;
export const commentsCountArray = [{id:1, count: 10}, {id: 2, count:20}];

export const comments = [{
  id: 1,
  userName: 'a',
  advantages: 'хороший звук',
  disadvantages: 'не очень качественный корпус',
  comment: 'В целом, понравилось',
  rating: 5,
  createAt: '',
  guitarId: 100,
}];
