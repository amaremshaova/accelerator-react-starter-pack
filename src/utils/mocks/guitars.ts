import { GuitarType } from '../../components/filters/filters';
import { Guitar } from '../../types/guitar';

export const guitars : Guitar[] = [
  {
    id: 1,
    name: 'Честер Bass',
    vendorCode: 'SO757575',
    type: GuitarType.Electric,
    description: 'Замечательный малобюджетный вариант, созданный для новичков, которые отдают предпочтение мелодичным стилям. Прекрасно звучат блюз и баллады, исполненные на этой гитаре. Акустические свойства весьма высоки, в отличие от ее стоимости.',
    previewImg: 'img/guitar-1.jpg',
    stringCount: 7,
    rating: 3.5,
    price: 17500,
  },
  {
    id: 2,
    name: 'Честер Z300',
    vendorCode: 'TK129049',
    type: GuitarType.Electric,
    description: 'Эргономичность гитары и качество сборки являются, пожалуй, её главными преимуществами. Идеальное расположение в руках музыканта дополняется прочностью конструкции из клёна.',
    previewImg: 'img/guitar-8.jpg',
    stringCount: 7,
    rating: 3.5,
    price: 29500,
  }];


export const guitarsCount = 2;

export const minPrice  = 17500;
export const maxPrice  = 29500;
