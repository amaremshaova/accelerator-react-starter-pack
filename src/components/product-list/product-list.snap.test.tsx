import {render} from '@testing-library/react';
import {BrowserRouter, Navigate, Route, Routes} from 'react-router-dom';
import { Provider } from 'react-redux';
import { guitars, maxPrice, minPrice } from '../../utils/mocks/guitars';
import { commentsCountArray } from '../../utils/mocks/comments';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { AppRoute } from '../../const';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import Catalog from '../catalog-page/catalog-page';
import ProductPage from '../product-page/product-page';
import NotFoundPage from '../not-found-page/not-found-page';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  DATA: {guitars: guitars,
    likeGuitars: guitars,
    guitarsCount: 27,
    minPrice: minPrice,
    maxPrice: maxPrice,
    commentsCount: commentsCountArray},
  APP : {productsInCart: [{product: guitars[0], count: 1}]},
});

const history = createMemoryHistory();

describe('Component: ProductList', () => {
  it('should render correctly', () => {
    history.push(AppRoute.Catalog + 1);
    const {container} = render(
      <Provider store = {store}>
        <BrowserRouter>
          <Routes>
            <Route path={AppRoute.Root} element={<Navigate to={AppRoute.CatalogStartPage} />}/>
            <Route path={`${AppRoute.CatalogPage  }:id`} element={<Catalog />}/>
            <Route path={`${AppRoute.Catalog  }:id`} element={<ProductPage />}/>
            <Route path={AppRoute.Undefined} element={<NotFoundPage/>}/>
          </Routes>
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
