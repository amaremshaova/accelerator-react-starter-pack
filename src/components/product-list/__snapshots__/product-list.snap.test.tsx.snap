// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProductList should render correctly 1`] = `
<div>
  <header
    class="header"
    id="header"
  >
    <div
      class="container header__wrapper"
    >
      <a
        class="header__logo logo"
        href="/catalog/page_1"
      >
        <img
          alt="Логотип"
          class="logo__img"
          height="70"
          src="./img/svg/logo.svg"
          width="70"
        />
      </a>
      <nav
        class="main-nav"
      >
        <ul
          class="main-nav__list"
        >
          <li>
            <a
              class="link main-nav__link"
              href="/catalog/page_1"
            >
              Каталог
            </a>
          </li>
          <li>
            <a
              class="link main-nav__link"
              href="/catalog/page_1"
            >
              Где купить?
            </a>
          </li>
          <li>
            <a
              class="link main-nav__link"
              href="/catalog/page_1"
            >
              О компании
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="form-search"
      >
        <form
          class="form-search__form"
          data-testid="form-search__form"
        >
          <button
            class="form-search__submit"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="form-search__icon"
              height="15"
              width="14"
            >
              <use
                href="#icon-search"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              Начать поиск
            </span>
          </button>
          <input
            autocomplete="off"
            class="form-search__input"
            data-testid="search"
            id="search"
            placeholder="что вы ищете?"
            type="text"
            value=""
          />
          <label
            class="visually-hidden"
            for="search"
          >
            Поиск
          </label>
        </form>
        <ul
          class="form-search__select-list
        hidden"
        >
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            <a
              class="form-search__link-guitar"
              href="/catalog/1"
            >
              Честер Bass
            </a>
          </li>
          <li
            class="form-search__select-item"
            tabindex="0"
          >
            <a
              class="form-search__link-guitar"
              href="/catalog/2"
            >
              Честер Z300
            </a>
          </li>
        </ul>
      </div>
      <a
        aria-label="Корзина"
        class="header__cart-link"
        href="/cart"
      >
        <svg
          aria-hidden="true"
          class="header__cart-icon"
          height="14"
          width="14"
        >
          <use
            href="#icon-basket"
          />
        </svg>
        <span
          class="visually-hidden"
        >
          Перейти в корзину
        </span>
        <span
          class="header__cart-count"
        >
          2
        </span>
      </a>
    </div>
  </header>
  <main
    class="page-content"
  >
    <div
      class="container"
    >
      <h1
        class="page-content__title title title--bigger"
      >
        Каталог гитар
      </h1>
      <ul
        class="breadcrumbs page-content__breadcrumbs"
      >
        <li
          class="breadcrumbs__item"
        >
          <a
            class="link"
            href="/catalog/page_1"
          >
            Главная
          </a>
        </li>
        <li
          class="breadcrumbs__item"
        >
          <a
            class="link"
            href="/catalog/page_1"
          >
            Каталог
          </a>
        </li>
      </ul>
      <div
        class="catalog"
      >
        <form
          class="catalog-filter"
        >
          <h2
            class="title title--bigger catalog-filter__title"
          >
            Фильтр
          </h2>
          <fieldset
            class="catalog-filter__block"
          >
            <legend
              class="catalog-filter__block-title"
            >
              Цена, ₽
            </legend>
            <div
              class="catalog-filter__price-range"
            >
              <div
                class="form-input"
              >
                <label
                  class="visually-hidden"
                >
                  Минимальная цена
                </label>
                <input
                  data-testid="priceMin"
                  id="priceMin"
                  name="от"
                  placeholder="17500"
                  type="number"
                  value=""
                />
              </div>
              <div
                class="form-input"
              >
                <label
                  class="visually-hidden"
                >
                  Максимальная цена
                </label>
                <input
                  data-testid="priceMax"
                  id="priceMax"
                  name="до"
                  placeholder="29500"
                  type="number"
                  value=""
                />
              </div>
            </div>
          </fieldset>
          <fieldset
            class="catalog-filter__block"
          >
            <legend
              class="catalog-filter__block-title"
            >
              Тип гитар
            </legend>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                data-testid="acoustic"
                id="acoustic"
                name="acoustic"
                type="checkbox"
              />
              <label
                for="acoustic"
              >
                Акустические гитары
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                data-testid="electric"
                id="electric"
                name="electric"
                type="checkbox"
              />
              <label
                for="electric"
              >
                Электрогитары
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                data-testid="ukulele"
                id="ukulele"
                name="ukulele"
                type="checkbox"
              />
              <label
                for="ukulele"
              >
                Укулеле
              </label>
            </div>
          </fieldset>
          <fieldset
            class="catalog-filter__block"
          >
            <legend
              class="catalog-filter__block-title"
            >
              Количество струн
            </legend>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                data-testid="4-strings"
                id="4-strings"
                name="4-strings"
                type="checkbox"
              />
              <label
                for="4-strings"
              >
                4
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                data-testid="6-strings"
                id="6-strings"
                name="6-strings"
                type="checkbox"
              />
              <label
                for="6-strings"
              >
                6
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                data-testid="7-strings"
                id="7-strings"
                name="7-strings"
                type="checkbox"
              />
              <label
                for="7-strings"
              >
                7
              </label>
            </div>
            <div
              class="form-checkbox catalog-filter__block-item"
            >
              <input
                class="visually-hidden"
                data-testid="12-strings"
                id="12-strings"
                name="12-strings"
                type="checkbox"
              />
              <label
                for="12-strings"
              >
                12
              </label>
            </div>
          </fieldset>
        </form>
        <div
          class="catalog-sort"
        >
          <h2
            class="catalog-sort__title"
          >
            Сортировать:
          </h2>
          <div
            class="catalog-sort__type"
          >
            <button
              aria-label="по цене"
              class="catalog-sort__type-button
          "
            >
              по цене
            </button>
            <button
              aria-label="по популярности"
              class="catalog-sort__type-button
          "
            >
              по популярности
            </button>
          </div>
          <div
            class="catalog-sort__order"
          >
            <button
              aria-label="По возрастанию"
              class="catalog-sort__order-button catalog-sort__order-button--up
          catalog-sort__order-button--active"
              data-testid="button-up"
            />
            <button
              aria-label="По убыванию"
              class="catalog-sort__order-button catalog-sort__order-button--down
          "
              data-testid="button-down"
            />
          </div>
        </div>
        <div
          class="cards catalog__cards"
        >
          <p
            style="margin-top: 200px; font-size: 30px;"
          >
            Загрузка товаров...
          </p>
        </div>
        <div
          class="pagination page-content__pagination"
        >
          <ul
            class="pagination__list"
          >
            
            <li
              class="pagination__page pagination__page--active"
            >
              <a
                class="link pagination__page-link"
                href="#"
              >
                1
              </a>
            </li>
            <li
              class="pagination__page "
            >
              <a
                class="link pagination__page-link"
                href="#"
              >
                2
              </a>
            </li>
            <li
              class="pagination__page "
            >
              <a
                class="link pagination__page-link"
                href="#"
              >
                3
              </a>
            </li>
            <li
              class="pagination__page pagination__page--next"
              id="next"
            >
              <a
                class="link pagination__page-link"
                href="#"
              >
                Далее
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </main>
  <footer
    class="footer"
  >
    <div
      class="footer__container container"
    >
      <a
        class="footer__logo logo"
        href="#"
      >
        <img
          alt="Логотип"
          class="logo__img"
          height="70"
          src="./img/svg/logo.svg"
          width="70"
        />
      </a>
      <div
        class="socials footer__socials"
      >
        <ul
          class="socials__list"
        >
          <li
            class="socials-item"
          >
            <a
              aria-label="facebook"
              class="socials__link"
              href="https://www.facebook.com/"
            >
              <svg
                aria-hidden="true"
                class="socials__icon"
                height="24"
                width="24"
              >
                <use
                  href="#icon-facebook"
                />
              </svg>
            </a>
          </li>
          <li
            class="socials-item"
          >
            <a
              aria-label="instagram"
              class="socials__link"
              href="https://www.instagram.com/"
            >
              <svg
                aria-hidden="true"
                class="socials__icon"
                height="24"
                width="24"
              >
                <use
                  href="#icon-instagram"
                />
              </svg>
            </a>
          </li>
          <li
            class="socials-item"
          >
            <a
              aria-label="twitter"
              class="socials__link"
              href="https://www.twitter.com/"
            >
              <svg
                aria-hidden="true"
                class="socials__icon"
                height="24"
                width="24"
              >
                <use
                  href="#icon-twitter"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <section
        class="footer__nav-section footer__nav-section--info"
      >
        <h2
          class="footer__nav-title"
        >
          О нас
        </h2>
        <p
          class="footer__nav-content footer__nav-content--font-secondary"
        >
          Магазин гитар, музыкальных инструментов и гитарная мастерская 
          <br />
           в Санкт-Петербурге.
          <br />
          <br />
          Все инструменты проверены, отстроены 
          <br />
           и доведены до идеала!
        </p>
      </section>
      <section
        class="footer__nav-section footer__nav-section--links"
      >
        <h2
          class="footer__nav-title"
        >
          Информация
        </h2>
        <ul
          class="footer__nav-list"
        >
          <li
            class="footer__nav-list-item"
          >
            <a
              class="link"
              href="#top"
            >
              Где купить?
            </a>
          </li>
          <li
            class="footer__nav-list-item"
          >
            <a
              class="link"
              href="#top"
            >
              Блог
            </a>
          </li>
          <li
            class="footer__nav-list-item"
          >
            <a
              class="link"
              href="#top"
            >
              Вопрос - ответ
            </a>
          </li>
          <li
            class="footer__nav-list-item"
          >
            <a
              class="link"
              href="#top"
            >
              Возврат
            </a>
          </li>
          <li
            class="footer__nav-list-item"
          >
            <a
              class="link"
              href="#top"
            >
              Сервис-центры
            </a>
          </li>
        </ul>
      </section>
      <section
        class="footer__nav-section footer__nav-section--contacts"
      >
        <h2
          class="footer__nav-title"
        >
          Контакты
        </h2>
        <p
          class="footer__nav-content"
        >
          Санкт-Петербург
          ,
          <br />
          м. Невский проспект
          ,  
          <br />
          ул. Казанская 6
        </p>
        <div
          class="footer__nav-content"
        >
          <svg
            aria-hidden="true"
            class="footer__icon"
            height="8"
            width="8"
          >
            <use
              href="#icon-phone"
            />
          </svg>
          <a
            class="link"
            href="tel:8-812-500-50-50"
          >
            8-812-500-50-50
             
          </a>
        </div>
        <p
          class="footer__nav-content"
        >
          Режим работы:
          <br />
          <span
            class="footer__span"
          >
            <svg
              aria-hidden="true"
              class="footer__icon"
              height="13"
              width="13"
            >
              <use
                href="#icon-clock"
              />
            </svg>
            <span>
              с 11:00 до 20:00
               
            </span>
            <span>
              без выходных
            </span>
          </span>
        </p>
      </section>
    </div>
  </footer>
</div>
`;
